tests/test_registercloudguest.py

```diff
@@ -2108,14 +2108,18 @@ def test_register_cloud_baseprod_ok_recommended_extensions_failed(
         'plugin:/susecloud?credentials=Basesystem_Module_x86_64&'
         'path=/repo/SUSE/Updates/SLE-Module-Basesystem/15-SP4/x86_64/update/'
     )
-    mock_exec_subprocess.side_effect = [True, False]
+    findmnt_return = b'{"filesystems": [{"target": "/","source": ' + \
+        b'"/dev/sda3","fstype": "xfs","options": "rw"}]}'
+    mock_exec_subprocess.return_value = findmnt_return, b'', 0
```

Comment:
I added a test based on the real output of findmnt

Reply:
interesting that it works on py 3.11 but seems to fail on 3.6

Reply:
Thanks for the changes, AFAICT it's missing the lines inside the if
```python
        if utils.is_transactional_system():
            activate_prod_cmd = 'transactional-update register -p {} '
            activate_prod_cmd += '-r ADDITIONAL REGCODE'
```
because the options of filesystems need a `ro`

```python
else:
        fsinfo = json.loads(output.decode())
        fsdata = fsinfo.get('filesystems')
        if fsdata:
            fsoptions = fsdata[0].get('options')
            # If we are on a RO system we need to use the
            # transactional-update command
            if 'ro' in fsoptions.split(','):
```

Reply:
will take a look at that test

Reply:
the `ro` change works